.github/workflows/main.yml

```diff
@@ -0,0 +1,53 @@
+name: Build
+
+on:
+  push:
+    branches: ["main"]
+  pull_request:
+    branches: ["main"]
```

Comment:
Since you are using `upload-release-action` step, I doubt it will work without a ref to a release. So I think it is better to change to something like on tag creation or on release creation.

Example:
```yaml
on:
  release:
    types: [published]
```